middlewares/validators/userStatus.js

```diff
@@ -2,20 +2,31 @@ const Joi = require("joi");
 const { userState } = require("../../constants/userStatus");
 const threeDaysInMilliseconds = 172800000;
 
-const validateUpdatedUserStatus = async (req, res, next) => {
+const validateUserStatusData = async (todaysTime, req, res, next) => {
   const schema = Joi.object({
     currentStatus: Joi.object().keys({
-      state: Joi.string().trim().valid(userState.IDLE, userState.ACTIVE, userState.OOO),
+      state: Joi.string()
+        .trim()
+        .valid(userState.IDLE, userState.ACTIVE, userState.OOO)
+        .error(new Error(`Invalid State. State must be either IDLE, ACTIVE or OOO`)),
       updatedAt: Joi.number().required(),
-      from: Joi.number().required(),
+      from: Joi.number()
+        .min(todaysTime)
+        .required()
+        .error(new Error(`The value for the 'from' field must be a date that is after the today's date.`)),
       until: Joi.any().when("state", {
         is: userState.OOO,
-        then: Joi.number().required(),
+        then: Joi.number()
+          .min(Joi.ref("from"))
+          .required()
+          .error(new Error(`The value for the 'until' field must be a date that is after the 'from' date.`)),
```

Comment:
what if I am OOO only for today

Reply:
yeah it will work even if you mark yourself OOO for a day. I'll revise the message to make it more appropriate.